lib/facter/framework/core/fact_loaders/fact_loader.rb

```diff
@@ -21,7 +21,12 @@ def initialize
     def load(user_query, options)
       @internal_facts = load_internal_facts(user_query, options)
       @custom_facts = load_custom_facts(options)
-      @external_facts = load_external_facts(options)
+      if ENV['INSIDE_FACTER']
```

Comment:
@joshcooper  I have reworked this but seem to be running into errors with `ENV['INSIDE_FACTER']` leaking between spec tests; it always leak in the same way, which is good. I think it happens when we mock external facts, like this(but still digging into the root cause):
```
      allow(external_fact_loader_double).to receive(:external_facts).and_return([loaded_env_custom_fact])
```

Reply:
OK, I just deleted the ENV var after we set it during external fact collection, and it seems to solve my leaky ENV vars.